lib/bson/binary.rb

```diff
@@ -229,7 +230,7 @@ def to_uuid(representation = nil)
           hex
         else
           raise ArgumentError, "Invalid representation: #{representation}"
-        end.sub(/(.{8})(.{4})(.{4})(.{12})/, '\1-\2-\3-\4')
+        end.sub(/(.{8})(.{4})(.{4})(.{4})(.{12})/, '\1-\2-\3-\4-\5')
```

Comment:
This should also have the anchors added.

Reply:
Updated